File: app/src/main/java/com/lucas/omnia/adapters/PartyAdapter.java
package com.lucas.omnia.adapters;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.lucas.omnia.R;
import com.lucas.omnia.models.Party;

import java.util.List;

public class PartyAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {

    private final List<Party> partyList;

    public static class PartyViewHolder extends RecyclerView.ViewHolder {
        public TextView labelTextView;
        public TextView nameTextView;

        PartyViewHolder(View itemView) {
            super(itemView);

            labelTextView = itemView.findViewById(R.id.party_tv_label_body);
            nameTextView = itemView.findViewById(R.id.party_tv_name_body);
        }
    }

    public PartyAdapter(List<Party> partyList) {
        this.partyList = partyList;
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
        return new PartyViewHolder(inflater.inflate(R.layout.item_party, parent, false));
    }

    @Override
    public void onBindViewHolder(RecyclerView.ViewHolder viewHolder, int position) {
        Party party = partyList.get(position);
        TextView labelTextView = ((PartyViewHolder) viewHolder).labelTextView;
        TextView nameTextView = ((PartyViewHolder) viewHolder).nameTextView;


        labelTextView.setText(party.getLabel());
        nameTextView.setText(party.getName());
    }

    @Override
    public int getItemCount() {
        return partyList.size();
    }
}

File: app/src/main/java/com/lucas/omnia/adapters/PartyAdapter.kt
package com.lucas.omnia.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.lucas.omnia.R
import com.lucas.omnia.models.Party

class PartyAdapter(private val partyList: List<Party>) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
    class PartyViewHolder internal constructor(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var labelTextView: TextView = itemView.findViewById(R.id.party_tv_label_body)
        var nameTextView: TextView = itemView.findViewById(R.id.party_tv_name_body)

    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        return PartyViewHolder(inflater.inflate(R.layout.item_party, parent, false))
    }

    override fun onBindViewHolder(viewHolder: RecyclerView.ViewHolder, position: Int) {
        val (label, name) = partyList[position]
        val labelTextView = (viewHolder as PartyViewHolder).labelTextView
        val nameTextView = viewHolder.nameTextView
        labelTextView.text = label
        nameTextView.text = name
    }

    override fun getItemCount(): Int {
        return partyList.size
    }
}